bids/cli.py

```diff
@@ -0,0 +1,102 @@
+from pathlib import Path
+import click
+
+from . import __version__
+from .layout import BIDSLayoutIndexer, BIDSLayout
+from .utils import validate_multiple as _validate_multiple
+
+
+# alias -h to trigger help message
+CONTEXT_SETTINGS = {'help_option_names': ['-h', '--help']}
+
+
+class PathOrRegex(click.ParamType):
+    "A helper Type to parse BIDSLayoutIndexer ignore/force entries"
+    name = "path or m/regex/"
+
+    def convert(self, value, param, ctx):
+        if value.startswith('m/') and value.endswith('/'):
+            # regex pattern
+            import re
+            value = re.compile(value[2:-1])
+        # otherwise, return as is
```

Comment:
Excess comments, IMO. Let's also insist that it actually have the full form, rather than interpreting `m/` as a regex.

```suggestion
        import re
        if re.match(r"^m/.*/$", value):  # has form "m/<regex>/"
            value = re.compile(value[2:-1])
```